### ISSUE TITLE
Plugin: Translations

### ISSUE THREAD

**Issue by olleharstedt** [SamMousa/ls3-question-objects-spec#8]
More of a question, but what's the policy for plugins and translations? How would a translator translate a plugin? Is that wanted, even?

**Comment by SamMousa** [comment]
I think that's definitely wanted, @c-schmitz you agree that it's wanted even if we don't know how yet?

From practical point of view, we can support a gettext category for each plugin and then use the normal yii translation stuff. 
http://www.yiiframework.com/doc/guide/1.1/en/topics.i18n

**Comment by Shnoulle** [comment]
Linked issue(s) in mantis : 

- https://bugs.limesurvey.org/view.php?id=9650 : Extend lang : add new string for this plugin
- https://bugs.limesurvey.org/view.php?id=9649 : Update core string 

The 2 concept are really different. Think the system must be in API, not in event.

**Comment by SamMousa** [comment]
@Shnoulle I agree. Not sure yet how though.

**Comment by olleharstedt** [comment]
We don't want to include translation files with the plugins, I suppose?

**Comment by Shnoulle** [comment]
@olleharstedt : i think we need to allow plugin to add translation file. And in different way seems better.

- Using an Array in a php file : http://www.yiiframework.com/doc/api/1.1/CPhpMessageSource
- Using a po system : http://www.yiiframework.com/doc/api/1.1/CGettextMessageSource
- Using DB : http://www.yiiframework.com/doc/api/1.1/CDbMessageSource

And this must have a "this language is in plugin or not" fix. For example : i can realease a plugin in french+english but not in german. If a German user want it : he can contribute on my plugin (or fork it).

You don't think ?

**Comment by olleharstedt** [comment]
Yes, that makes sence. The important thing is that the translate system is as in core, so it's easy to learn for core translators.
